src/packagedcode/spec.py

```diff
@@ -0,0 +1,156 @@
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from collections import OrderedDict
+import csv
+import glob
+import io
+import logging
+import os
+import re
+
+from gemfileparser import GemfileParser
+
+"""
+Handle Cocoapods(.podspec) and Ruby(.gemspec) files.
+"""
+
+TRACE = False
+
+logger = logging.getLogger(__name__)
+
+if TRACE:
+    import sys
+    logging.basicConfig(stream=sys.stdout)
+    logger.setLevel(logging.DEBUG)
+
+
+class ParseSpec(object):
+    def parse_spec(self, location):
+        """
+        Return dictionary contains podspec or gemspec file data.
+        """
+        with io.open(location, encoding='utf-8', closefd=True) as data:
+            lines = data.readlines()
+
+        spec_data = {}
+
+        for line in lines:
+            if '.name' in line:
+                name = re.sub(r'/*.*name.*?=', '', line)
+                spec_data['name'] = get_stripped_data(name)
+            elif '.version' in line and '.version.' not in line:
+                version = re.sub(r'/*.*version.*?=', '', line)
+                spec_data['version'] = get_stripped_data(version)
+            elif '.license' in line:
+                license_type = re.sub(r'/*.*license.*?=', '', line)
+                spec_data['license'] = get_stripped_data(license_type)
+            elif '.source' in line and '.source_files' not in line:
+                source = re.sub(r'/*.*source.*?>', '', line)
+                stripped_source = re.sub(r',.*', '', source)
+                spec_data['source'] = get_stripped_data(stripped_source)
+            elif '.email' in line:
+                emails = re.sub(r'/*.*email.*?=', '', line)
+                stripped_emails = get_stripped_data(emails)
+                stripped_emails = stripped_emails.strip()
+                stripped_emails = stripped_emails.split(',')
+                spec_data['email'] = stripped_emails
+            elif '.author' in line:
+                authors = re.sub(r'/*.*author.*?=', '', line)
+                stripped_authors = get_stripped_data(authors)
+                stripped_authors = stripped_authors.replace(' => ', "=>")
+                stripped_authors = stripped_authors.strip()
+                stripped_authors = stripped_authors.split(',')
+                spec_data['author'] = stripped_authors
+            elif '.summary' in line:
+                summary = re.sub(r'/*.*summary.*?=', '', line)
```

Comment:
Could a simpler `line.rpartition('=')` be all you need here and elsewhere?

Reply:
In parsing gemspec there is a line
`line = 'spec.summary = %q{The loading cat generator.}'`
To get desire data of summary we can use

1. summary = re.sub(r'/*.*summary.*?=', '', line) (which I used)
2. summary = line.rpartition('=') (you suggested)
3. summary = line.split('=')
and many more ways
So what should I use??